subsys/net/ip/net_context.c

```diff
@@ -1229,32 +1229,39 @@ static struct net_pkt *context_alloc_pkt(struct net_context *context,
 					 size_t len, s32_t timeout)
 {
 	struct net_pkt *pkt;
+	struct net_if *interface = net_context_get_iface(context);
+	sa_family_t family = net_context_get_family(context);
+	u16_t ip_proto = net_context_get_ip_proto(context);
 
 #if defined(CONFIG_NET_CONTEXT_NET_PKT_POOL)
 	if (context->tx_slab) {
-		pkt = net_pkt_alloc_from_slab(context->tx_slab(), timeout);
-		if (!pkt) {
-			return NULL;
-		}
+		struct k_mem_slab *slab = context->tx_slab();
 
-		net_pkt_set_iface(pkt, net_context_get_iface(context));
-		net_pkt_set_family(pkt, net_context_get_family(context));
-		net_pkt_set_context(pkt, context);
-
-		if (net_pkt_alloc_buffer(pkt, len,
-					 net_context_get_ip_proto(context),
-					 timeout)) {
-			net_pkt_unref(pkt);
-			return NULL;
+		k_mutex_unlock(&context->lock);
+		pkt = net_pkt_alloc_from_slab(slab, timeout);
+		if (pkt) {
+			net_pkt_set_iface(pkt, interface);
+			net_pkt_set_family(pkt, family);
+			net_pkt_set_context(pkt, context);
+
+			if (net_pkt_alloc_buffer(pkt, len,
```

Comment:
Please use `< 0` for testing for errors.

Reply:
This line has merely been reindented after the block was wrapped in `if (pkt)`.

Reply:
Ok, maybe it was re-indented, but the patch as a whole does fix obscure issues with the code. And that's just another maybe not too-much, but still an obscure issue. It's an issue because: a) We use `< 0` or ` == NULL` / `!ptr` for error checking in the code. In this regard, using `if (foo)` to catch an error looks pretty confusing; b) MISRA C, which we try to adopt, requires `if`, etc. conditions to be proper boolean expressions.

So, as long as that code appears as effected by your patch, please consider applying the suggested fix.